### PR TITLE
firstboot: make restart of vconsole-setup synchronuous 

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/firstboot/firstboot.c [hunk 1]
```diff
@@ -1598,10 +1602,26 @@ static int reload_vconsole(sd_bus **bus) {
                         return bus_log_connect_error(r, BUS_TRANSPORT_LOCAL);
         }
 
-        r = bus_call_method(*bus, bus_systemd_mgr, "RestartUnit", &error, NULL, "ss", "systemd-vconsole-setup.service", "replace");
+        r = bus_wait_for_jobs_new(*bus, &w);
+        if (r < 0)
+                return log_oom();
```
Comment (YHNdnzj) [inline src/firstboot/firstboot.c]: This can also fail while matching the signals.

File: src/firstboot/firstboot.c [hunk 2]
```diff
@@ -1598,10 +1602,26 @@ static int reload_vconsole(sd_bus **bus) {
...
+                            "ss", "systemd-vconsole-setup.service", "replace");
         if (r < 0)
                 return log_error_errno(r, "Failed to issue method call: %s", bus_error_message(&error, r));
-        log_info("Requested vconsole setup to apply key map configuration.");
+
+        r = sd_bus_message_read(reply, "o", &object);
+        if (r < 0)
+                return bus_log_parse_error(r);
+
+        r = bus_wait_for_jobs_add(w, object);
```
Comment (YHNdnzj) [inline src/firstboot/firstboot.c]: Just use `bus_wait_for_jobs_one`?

**Review by yuwata — APPROVED** [review]
LGTM.

**Comment by keszybz** [comment]
Thanks. I added a third commit to change the error handling for `bus_wait_for_jobs_new()` in the other few places that got it wrong.

**Pull request merged**